src/index.js

```diff
@@ -799,8 +799,18 @@ export default class Gantt {
     get_all_dependent_tasks(task_id) {
         let out = [];
         let to_process = [task_id];
+
+        //counter for endless loop check
+        let counter = 0;
         while (to_process.length) {
+        	
             const deps = to_process.reduce((acc, curr) => {
+            	// break if the loop is endless
+                counter++;
+            	if(counter === 1000) {
```

Comment:
1000 is an arbitrary number. It is not the best way for handling the recursion.
The recursion should be handled by the user it self while creating the task.